cellmlmanip/parser.py

```diff
@@ -406,3 +415,21 @@ def _parent_of(parent_name, child_name):
 
         raise ValueError('Cannot determine the source & target for connection (%s, %s) - (%s, %s)' %
                          (comp_1, var_1, comp_2, var_2))
+
+    def _validate(self, parser, tree):
+        """
+        Validates the given lxml ``tree`` against the CellML 1.0 RELAX NG schema.
+
+        :param parser: An `lxml.etree.XMLParser`
+        :param tree: An `lxml.etree.ElementTree` made with the given parser.
+        """
+
+        # Create RelaxNG object
```

Comment:
If a parser might be used for several models, should this be instantiated in the constructor and saved? Very minor point though!

Reply:
Regarding that: What do you think of parser objects, that are mostly single-use? I always design them as classes but only because it groups my code nicely and I feel a bit dirty writing a bunch of private methods in module scope. I looked around a bit and parsers are usually classes (with non-static methods), although example code only uses them once.

Off-topic I know :D

Reply:
Actually, that is relevant here because this parser class can only be used once :D

Reply:
I normally make them classes too. A class that only gets used once is still fine! Of course in some languages everything is a class...